tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala

```diff
@@ -33,7 +53,93 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
   }
 
   def addMultiSampleJobs() = {
+    val initRefMap = samples map { x => x._2 -> getReferenceSamplesForSample(x._1) }
+    initRefMap.values.filter(_.isLeft).foreach {
+      case -\/(message) => Logging.addError(message)
+      case _            => ;
+    }
+
+    val refMap: Map[Sample, Set[Sample]] = initRefMap.filter(_._2.isRight).map { x =>
```

Comment:
Similar to the above, I think using `.collect` and pattern matching would make this more clear.